### PR TITLE
fix(core): miri tweaks

### PR DESCRIPTION
**Pull request by mmastrac**
- Fix some sketchiness in FastString that we were waiting on rusty_v8 to fix.
 - Run more tests under miri, comments for those that can't

### CODE REVIEW

File: .github/workflows/ci-coverage/action.yml [hunk 1]
```diff
@@ -12,4 +12,4 @@ runs:
       uses: codecov/codecov-action@v3
       with:
         files: lcov.info
-        fail_ci_if_error: true
```
Comment (mmastrac) [inline .github/workflows/ci-coverage/action.yml:15]: Codecov is not always succeeding :/

File: core/fast_string.rs [hunk 2]
```diff
@@ -471,9 +451,6 @@ macro_rules! __op_name_fast {
 mod tests {
   use super::*;
 
-  // TODO(mmastrac): we need to upstream as_str() on OneByteConst to rusty_v8 to get
-  // rid of the hacks that break miri.
-  #[cfg(not(miri))]
```
Comment (bartlomieju) [inline core/fast_string.rs:476]: Have we actually upstreamed this change and just waiting on `rusty_v8` release?

File: core/module_specifier.rs [hunk 3]
```diff
@@ -335,7 +335,11 @@ mod tests {
     ];
 
     // The local path tests assume that the cwd is the deno repo root.
-    let cwd = current_dir().unwrap();
+    let cwd = if cfg!(miri) {
+      PathBuf::from("/miri")
+    } else {
+      current_dir().unwrap()
+    };
```
Comment (bartlomieju) [inline core/module_specifier.rs:342]: Add a comment here?

File: core/fast_string.rs (hunk 2 above)
Comment (mmastrac) [reply core/fast_string.rs:476]: It's been out for a bit -- I just forgot about this cleanup.

File: core/module_specifier.rs [hunk 4]
```diff
@@ -335,7 +335,11 @@ mod tests {
     ];
 
     // The local path tests assume that the cwd is the deno repo root.
```
Comment (mmastrac) [inline core/module_specifier.rs:337]:
```suggestion
    // The local path tests assume that the cwd is the deno repo root. Note
    // that we can't use `cwd` in miri tests, so we just use `/miri` instead.
```

File: core/fast_string.rs [hunk 5]
```diff
@@ -29,39 +29,19 @@ impl FastStaticString {
...
-    // SAFETY: Workaround compiler error in create_external_onebyte_const for long ascii strings
-    unsafe {
-      debug_assert!(s.is_ascii());
-      let c = v8::String::create_external_onebyte_const(&[]);
-      let mut ptrs: OneByteConst = std::mem::transmute(c);
-      ptrs.cached_data = std::mem::transmute(s.as_ptr());
-      ptrs.length = s.len();
-      std::mem::transmute(ptrs)
-    }
+    v8::String::create_external_onebyte_const(s)
```
Comment (littledivy) [inline core/fast_string.rs:44]: nice

**Review by littledivy — APPROVED** [review]
lgtm

**Pull request merged**